src/main/java/org/broadinstitute/hellbender/utils/clipping/ReadClipper.java

```diff
@@ -488,10 +492,146 @@ protected GATKRead hardClipByReferenceCoordinates(final int refStart, final int
         if ( start > 0 && stop < read.getLength() - 1) {
             throw new GATKException(String.format("Trying to clip the middle of the read: start %d, stop %d, cigar: %s", start, stop, read.getCigar().toString()));
         }
-
         this.addOp(new ClippingOp(start, stop));
         final GATKRead clippedRead = clipRead(ClippingRepresentation.HARDCLIP_BASES);
         this.ops = null;
         return clippedRead;
     }
+
+
+
+    /**
+     * Soft clip the read to the variable region (from refStart to refStop) processing also the clipped bases
+     *
+     * @param read     the read to be clipped
+     * @param refStart the beginning of the variant region (inclusive)
+     * @param refStop  the end of the variant region (inclusive)
+     * @return the read soft clipped to the variant region
+     */
+    public static GATKRead softClipToRegionIncludingClippedBases( final GATKRead read, final int refStart, final int refStop ) {
+        final int start = read.getUnclippedStart();
+        final int stop = start + CigarUtils.countRefBasesBasedOnCigar(read, 0, read.numCigarElements()) - 1;
+
+        if (start <= refStop && stop >= refStart) {
+            if (start < refStart && stop > refStop) {
+                return (new ReadClipper(read)).softClipBothEndsByReferenceCoordinates(refStart - 1, refStop + 1);
+            } else if (start < refStart) {
+                return (new ReadClipper(read)).softClipByReferenceCoordinates(-1, refStart - 1);
+            } else if (stop > refStop) {
+                return (new ReadClipper(read)).softClipByReferenceCoordinates(refStop + 1, -1);
+            }
+            return read;
+        } else {
+            return ReadUtils.emptyRead(read);
```

Comment:
I agree with Megan that this should probably emit a warning when this happens since I don't think it's something anyone would expect to happen.